tests/baselines/reference/wideningTuples2.errors.txt

```diff
@@ -0,0 +1,11 @@
+tests/cases/conformance/types/tuple/wideningTuples2.ts(2,9): error TS7005: Variable 'intermediate' implicitly has an '[any]' type.
+
+
+==== tests/cases/conformance/types/tuple/wideningTuples2.ts (1 errors) ====
+    var foo: () => [any] = function bar() {
+        let intermediate = bar();

```

Comment:
Yeah, as stated in the meeting I do think this is not a reasonable new error in a quite old test (and was dealt with in the prior iteration of this change), and this can be somewhat easily avoided by using the old behavior for named function expressions (which are generally only used when you want recursive references like this anyhow). A self-call witnessing the unwidened type is _very_ odd, given our widening rules (no exterior calls can witness the unwidened type). I think this can manifest in even _stranger_ behavior if you do something with the `bar` reference rather than just call it - like pass it into a function desiring a callback, eg `handle(bar)` where `declare function handle(a: () => [any]): void`, or worse yet `declare function handle<T>(a: T): T` where the unwidened type will simply propagate via inference - where then you'll get this widening error so far removed from anything that it's relevant to that it's quite clear something's gone wrong.